docs/ide/mcp.md

```diff
@@ -0,0 +1,166 @@
+
+## Finding MCP servers
+
+MCP's [official server repository](https://github.com/modelcontextprotocol/servers) is a great starting point for reference, official, and community-contributed servers that showcase MCP's versatility. You can explore servers for various functionalities, such as file system operations, database interactions, and web services.
+
+MCP is still a relatively new standard, and the ecosystem is rapidly evolving. As more developers adopt MCP, you can expect to see an increasing number of servers and tools available for integration with your projects.
+
+## Adding  an MCP server
+
+### Create a file to manage configuration of MCP servers
+
+If you do not already have an `mcp.json` file, you can create it in various locations depending on the repos, users, and IDEs you would like the servers to be available/used for.
+
+### File locations for automatic discovery of MCP configuration
+
+Visual Studio looks for MCP server configurations in the following directories, in the following order order:
+
+1.  `%USERPROFILE%\.mcp.json` serves as a global MCP server configuration for a specific user. Adding an MCP server here would make it load for all Visual Studio solutions. 
+1.  `<SOLUTIONDIR>\.vs\mcp.json` is specific to Visual Studio and only loads the specified MCP servers for a specific user, for the specified solution.
+
+If you are looking for an MCP configuration that you can track in source control for a repo, this option woudl work well:
+
+3.  `<SOLUTIONDIR>\.mcp.json` 
+
+Visual Studio will also check for MCP configurations other development environments have have set up. These are scoped to the repository/solution and are typically not source controlled.
+
+4.  `<SOLUTIONDIR>\.vscode\mcp.json`
+5.  `<SOLUTIONDIR>\.cursor\mcp.json`
+
+### MCP configuration format
+
+You may define both **remote** (URL + credentials) and **local** (command-line invocation).
+
+It’s common to invoke tools via package managers (e.g. `pip install … && mcp-server or npx @cursor/mcp`) - Visual Studio will respect whatever command you specify so you can pin versions or flags.
+
+Format must follow the MCP spec (e.g. array of server objects, each with `name`, `command` or `url`, `transport`, etc.).
+
+### Editing MCP configuration
+
+With an existing `mcp.json` file, add the file location to the Solution Items in Solution Explorer.
+
+When the file is saved with valid syntax, GitHub Copilot's agent will restart and reload the configured servers.
+
+![Copilot Agent Settings](./media/vs-2022/copilotagent-mcp/mcp-addsolutionitem.png)
+
+### Tool Lifecycle
+
+As soon as a server is discovered or added, Visual Studio initializes it (handshake + tool list).
+
+Visual Studio subscribes to the MCP event notifications/tools/list_changed.
+
+When that event fires, Visual Studio resets any prior acceptances/permissions on tools (to prevent “rug-pull” attacks), re-fetches the tool list, and updates the count/UI live.
```

Comment:
Is this an accurate statement? I may need to play with this, but I wasn't under the impression that we were resetting permissions on tools every time the list changes